src/testUtils.ts

```diff
@@ -166,9 +168,25 @@ export function goTest(testconfig: TestConfig): Thenable<boolean> {
 			const outBuf = new LineBuffer();
 			const errBuf = new LineBuffer();
 
-			outBuf.onLine(line => outputChannel.appendLine(expandFilePathInOutput(line, testconfig.dir)));
-			outBuf.onDone(last => last && outputChannel.appendLine(expandFilePathInOutput(last, testconfig.dir)));
+			const packageResultLineRE = /^(ok|FAIL)[ \t]+(.+?)[ \t]+([0-9\.]+s|\(cached\))/; // 1=ok/FAIL, 2=package, 3=time/(cached)
+			const testResultLines: string[] = [];
+
+			const processTestResultLine = (line: string) => {
+				testResultLines.push(line);
+				const result = line.match(packageResultLineRE);
+				if (result) {
+					const packageNameArr = result[2].split('/');
+					const baseDir = path.join(workspaceRoot, ...packageNameArr.slice(1));
```

Comment:
Using `workspaceRoot` doesn't work always here.

For example, assume I have opened the [golang example repo](https://github.com/golang/example/) in vscode and assume that a test in the `stringutil` package is failing

`workspaceRoot` will be `/Users/myGopath/src/github.com/golang/example`
`packageNameArr` after the slice will be [golang, example, stringutil]
`baseDir` becomes `/Users/myGopath/src/github.com/golang/example/golang/example/stringutil` which is wrong

`currentGoWorkspace` variable on the other hand has what you need. So use that instead
`const baseDir = path.join(currentGoWorkspace, ...packageNameArr);

 Note that `currentGoWorkspace` can be undefined when we cannot determine the gopath or if current file doesn't not belong to current workspace

Reply:
Fixed!